--- libsmb2-3.0.0/lib/init.c	2019-06-09 23:38:20.000000000 +0200
+++ libsmb2-3.0.0-patched/lib/init.c	2021-11-29 23:04:02.150371649 +0100
@@ -63,6 +63,11 @@
 #define getlogin() "Guest"
 #endif
 
+#ifdef __SWITCH__
+#include <errno.h>
+#define getlogin_r(a,b) ENXIO
+#endif // __SWITCH__
+
 static int
 smb2_parse_args(struct smb2_context *smb2, const char *args)
 {
@@ -214,7 +219,7 @@
         }
         memset(smb2, 0, sizeof(struct smb2_context));
 
-        smb2_set_user(smb2, getlogin());
+        smb2_set_user(smb2, "");
         smb2->fd = -1;
         smb2->sec = SMB2_SEC_UNDEFINED;
         smb2->version = SMB2_VERSION_ANY;
